src/applications/vaos/new-appointment/redux/actions.js

```diff
@@ -258,53 +264,165 @@ export function fetchFacilityDetails(facilityId) {
   };
 }
 
+export function checkEligibility(location, siteId) {
+  return async (dispatch, getState) => {
+    const state = getState();
+    const useVSP = vaosVSPAppointmentNew(state);
+    const directSchedulingEnabled = vaosDirectScheduling(state);
+    const typeOfCareId = getTypeOfCare(getState().newAppointment.data)?.id;
+
+    dispatch({
+      type: FORM_ELIGIBILITY_CHECKS,
+    });
+
+    try {
+      const eligibilityData = await getEligibilityData(
+        location,
+        typeOfCareId,
+        siteId,
+        directSchedulingEnabled,
+        useVSP,
+      );
+
+      recordEligibilityGAEvents(eligibilityData, typeOfCareId, siteId);
+      logEligibilityExplanation(eligibilityData, typeOfCareId, location.id);
+
+      dispatch({
+        type: FORM_ELIGIBILITY_CHECKS_SUCCEEDED,
+        typeOfCareId,
+        eligibilityData,
+      });
+
+      try {
+        const eligibility = getEligibilityStatus(getState());
+        if (!eligibility.direct && !eligibility.request) {
+          // Remove parse function when converting this call to FHIR service
+          const thunk = fetchFacilityDetails(location.id);
+          await thunk(dispatch, getState);
+        }
+      } catch (e) {
+        captureError(e);
+      }
+    } catch (e) {
+      captureError(e, false, 'facility page');
+      dispatch({
+        type: FORM_ELIGIBILITY_CHECKS_FAILED,
+      });
+    }
+  };
+}
+
 export function openFacilityPageV2(page, uiSchema, schema) {
   return async (dispatch, getState) => {
     try {
       const initialState = getState();
       const newAppointment = initialState.newAppointment;
-      const typeOfCareId = getTypeOfCare(newAppointment.data)?.id;
+      const data = newAppointment.data;
+      const typeOfCare = getTypeOfCare(newAppointment.data);
+      const typeOfCareId = typeOfCare.id;
       const userSiteIds = selectSystemIds(initialState);
       const useVSP = vaosVSPAppointmentNew(initialState);
       let parentFacilities = newAppointment.parentFacilities;
+      let locations = null;
+      let locationId = data.vaFacility;
 
       dispatch({
         type: FORM_PAGE_FACILITY_V2_OPEN,
       });
 
-      if (!parentFacilities) {
-        parentFacilities = await getOrganizations({
-          siteIds: userSiteIds,
-          useVSP,
+      try {
+        if (!parentFacilities) {
+          dispatch({ type: FORM_FETCH_PARENT_FACILITIES });
```

Comment:
Do we need both the parent and child actions for fetching the facility info? It doesn't seem like anything needs to happen in the UI when the parents finish loading? Or am I missing something?

Reply:
The only thing these actions do individually are set the loading statuses, which we use to display the loading indicator when the page first loads.  I supposed I could set both the parent and child facilities to loading with the page open action, and set both to `succeeded` on page load success? 

Either way this will be refactored with the upcoming PR to use the new eligibility endpoints.

Reply:
I was wondering if the new page could use one loading status, since they seem to always be fetched together now, mainly.

I agree though that if you're re-writing this action with the new endpoints, we should leave it alone for now.